src/IB/IBFEMethod.cpp

```diff
@@ -1393,37 +1258,8 @@ IBFEMethod::initializeFEEquationSystems()
         // NOTE: libMesh does not appear to preserve the type of the vector
         // after restart (GHOSTED vectors are now PARALLEL), and so we
         // manually reset these vectors here.
-        auto insert_parallel_into_ghosted = [](const PetscVector<Number>& parallel_vector,
-                                               PetscVector<Number>& ghosted_vector) {
-            TBOX_ASSERT(parallel_vector.size() == ghosted_vector.size());
-            TBOX_ASSERT(parallel_vector.local_size() == ghosted_vector.local_size());
-            ghosted_vector = parallel_vector;
-            ghosted_vector.close();
-        };
-
-        auto convert_parallel_to_ghosted = [&](const std::string& vector_name, System& system) {
-            std::unique_ptr<NumericVector<double> > clone_vector;
-            if (from_restart)
-            {
-                NumericVector<double>* current = system.request_vector(vector_name);
-                if (current != nullptr)
-                {
-                    clone_vector = current->clone();
-                }
-            }
-            system.remove_vector(vector_name);
-            system.add_vector(vector_name, /*projections*/ true, /*type*/ GHOSTED);
-
-            if (clone_vector != nullptr)
-            {
-                const auto& parallel_vector = dynamic_cast<const PetscVector<Number>&>(*clone_vector);
-                auto& ghosted_vector = dynamic_cast<PetscVector<Number>&>(system.get_vector(vector_name));
-                insert_parallel_into_ghosted(parallel_vector, ghosted_vector);
-            }
-        };
-
-        const std::array<std::string, 2> system_names{ { COORDS_SYSTEM_NAME, VELOCITY_SYSTEM_NAME } };
-        const std::array<std::string, 2> vector_names{ { "new", "half" } };
+        const std::vector<std::string> system_names{ COORDS_SYSTEM_NAME, VELOCITY_SYSTEM_NAME };
+        const std::vector<std::string> vector_names{ "new", "half" };
```

Comment:
Isn't all this stuff done in `FEMechanicsBase` now?

Reply:
Nope; see above.